### PR TITLE
[Azure.ResourceManager.MigrationDiscoverySap] Fix for TestDiscoveryOperations failure

### LINKED ISSUE #43073: [Azure.ResourceManager.MigrationDiscoverySap] TestDiscoveryOperations test is failing
https://dev.azure.com/azure-sdk/public/_build/results?buildId=3646971&view=ms.vss-test-web.build-test-results-tab&runId=46856532&resultId=103612&paneView=debug

```
Azure.RequestFailedException : Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.
RequestId:cbdc7c3b-201e-003f-803c-81608b000000
Time:2024-03-28T18:20:44.8998078Z
Status: 403 (Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.)
ErrorCode: AuthenticationFailed


Additional Information:
AuthenticationErrorDetail: Signature not valid in the specified time frame: Start [Thu, 28 Mar 2024 09:19:23 GMT] - Expiry [Thu, 28 Mar 2024 09:54:23 GMT] - Current [Thu, 28 Mar 2024 18:20:44 GMT]
```

### PR DESCRIPTION
**Pull request by shubhamgarg2175**
fixes https://github.com/Azure/azure-sdk-for-net/issues/43073

### CODE REVIEW

File: sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/tests/Tests/MigrationSapDiscoveryTests.cs [hunk 1]
```diff
@@ -146,13 +145,16 @@ public async Task PostImportEntities(ArmClient client, SapDiscoverySiteResource
         var operationStatusObj = JObject.Parse(opStatus.Content.ToString());
         var inputExcelSasUri = operationStatusObj?["properties"]?["discoveryExcelSasUri"].ToString();
 
-        //Upload here
-        using (FileStream stream = File.OpenRead(excelPathToImport))
+        if (SessionEnvironment.Mode != RecordedTestMode.Playback)
```
Comment (live1206) [inline sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/tests/Tests/MigrationSapDiscoveryTests.cs:148]:
```suggestion
        if (SessionEnvironment.Mode == RecordedTestMode.Record)
```
We only want this during recording, there is another model `None`

Comment (live1206) [inline sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/tests/Tests/MigrationSapDiscoveryTests.cs:148]: please add some comments explaining why we are adding this if, basically this is a data-plane client which can't simply interact with [HttpMockServer](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/mgmtcommon/TestFramework/Microsoft.Azure.Test.HttpRecorder/HttpMockServer.cs).

File: sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/tests/Tests/MigrationSapDiscoveryTests.cs [hunk 2]
```diff
@@ -146,13 +145,16 @@ public async Task PostImportEntities(ArmClient client, SapDiscoverySiteResource
         var operationStatusObj = JObject.Parse(opStatus.Content.ToString());
         var inputExcelSasUri = operationStatusObj?["properties"]?["discoveryExcelSasUri"].ToString();
```
Comment (live1206) [inline sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/tests/Tests/MigrationSapDiscoveryTests.cs:146]: should this be included into the recording run as well? while playback mode, this is not needed.

**Review by live1206 — CHANGES_REQUESTED** [review]
As offline mentioned, the SAS token in recording files should be sanitized as well.

File: sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/tests/Tests/MigrationSapDiscoveryTests.cs (hunk 1 above)
Comment (live1206) [reply sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/tests/Tests/MigrationSapDiscoveryTests.cs:148]: `none` is `live`, which needs to skip this as well

Comment (shubhamgarg2175) [reply sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/tests/Tests/MigrationSapDiscoveryTests.cs:148]: Done

File: sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/tests/Tests/MigrationSapDiscoveryTests.cs (hunk 2 above)
Comment (shubhamgarg2175) [reply sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/tests/Tests/MigrationSapDiscoveryTests.cs:146]: Done

**Review by live1206 — APPROVED** [review]

**Pull request merged**